context/app/static/js/stores/useFilesStore.ts

```diff
@@ -1,23 +1,54 @@
 import { create } from 'zustand';
 
 export type FileDisplayOption = 'all' | 'qa/qc' | 'data products';
+export type FileFilter = 'qa/qc' | 'data products';
 
 export interface FilesStore {
   filesToDisplay: FileDisplayOption;
+  selectedFileFilters: FileFilter[];
+  toggleFileFilter: (filter: FileFilter) => void;
+  // Legacy methods for backward compatibility
   toggleDisplayOnlyQaQc: () => void;
   toggleDisplayOnlyDataProducts: () => void;
 }
 
 const useFilesStore = create<FilesStore>((set) => ({
   filesToDisplay: 'all',
+  selectedFileFilters: [],
+  toggleFileFilter: (filter: FileFilter) => {
+    set((state) => {
+      const isSelected = state.selectedFileFilters.includes(filter);
+      const newFilters = isSelected
+        ? state.selectedFileFilters.filter((f) => f !== filter)
+        : [...state.selectedFileFilters, filter];
+
+      // Determine the display option based on selected filters
+      let filesToDisplay: FileDisplayOption = 'all';
+      if (newFilters.length === 1) {
+        filesToDisplay = newFilters[0];
+      } else if (newFilters.length > 1) {
+        // When multiple filters are selected, use the first one for display purposes
+        // This ensures folders stay expanded (since filesToDisplay !== 'all')
+        filesToDisplay = newFilters[0];
```

Comment:
Not sure I'm following the logic here. Should the `if` statement just be checking that length is non 0?